### PR TITLE
Improvements in GUI and refactoring

### PR DESCRIPTION
**Pull request by Devas**
Added info about current page to the main page;
Moved buttons to the center of the screen;
Added first and last page buttons;
The buttons are now invisible if the user is at the end of the paginations;
Buttons are now disabled when user reaches the last/first page of pagination;
The discount displayed is not calculated in frontend;
Moved search by one keyword to database query
Refactoring

### CODE REVIEW

File: rest-angular/src/app/app.module.ts [hunk 1]
```diff
@@ -9,7 +9,11 @@ import { MainComponent } from './main.component';
 import { AppRoutingModule } from './app-routing.module';
 import { BookDetailsComponent } from './book-details.component';
 import { BookDetailsService } from './book-details.service';
+import { Config } from './config/config';
 
+export function initConfig(config: Config){
+ return () => config.load()
```
Comment (piotr921) [inline rest-angular/src/app/app.module.ts]: should't use -> ?

Comment (agawda) [reply rest-angular/src/app/app.module.ts]: No, it's not possible. :v

File: springapp/src/main/java/com/javaacademy/robot/controllers/BookRestAPIController.java [hunk 2]
```diff
@@ -42,11 +40,11 @@ public BookRestAPIController(BookService bookService, BookSearch bookSearch) {
 
     @RequestMapping("/api/search")
     public List<BookDto> searchResults(@RequestParam(value = "query") String query) {
-        return bookSearch.search(query);
+        return bookService.searchOneKeyword(query.toLowerCase());
```
Comment (patkkka) [inline springapp/src/main/java/com/javaacademy/robot/controllers/BookRestAPIController.java]: 👍

File: springapp/src/main/resources/application.properties [hunk 3]
```diff
@@ -1,6 +1,6 @@
 spring.datasource.url=jdbc:postgresql://localhost:5432/test_db
 spring.datasource.username=postgres
-spring.datasource.password=12345
+spring.datasource.password=
```
Comment (patkkka) [inline springapp/src/main/resources/application.properties]: 👍

File: springapp/src/test/java/com/javaacademy/robot/controllers/BookRestAPIControllerTest.java [hunk 4]
```diff
@@ -49,45 +51,75 @@ public void shouldReturnOneBook() {
...
     @Test
     public void shouldReturnZeroBooks() {
         bookServiceMock = mock(BookService.class);
-        BookRestAPIController bookRestAPIController = new BookRestAPIController(bookServiceMock, null);
+        BookRestAPIController bookRestAPIController = new BookRestAPIController(bookServiceMock);
 
         when(bookServiceMock.getAllBookDtos()).thenReturn(Collections.emptyList());
 
-        assertEquals(bookRestAPIController.getBooksNumber(), 0);
+        assertEquals(0, bookRestAPIController.getBooksNumber());
```
Comment (patkkka) [inline springapp/src/test/java/com/javaacademy/robot/controllers/BookRestAPIControllerTest.java]: 👍

File: springapp/src/test/java/com/javaacademy/robot/controllers/BookRestAPIControllerTest.java [hunk 5]
```diff
@@ -49,45 +51,75 @@ public void shouldReturnOneBook() {
...
+        BookService bookServiceMock = mock(BookService.class);
+        BookRestAPIController bookRestAPIController = new BookRestAPIController(bookServiceMock);
         String givenQuery = "Franklin";
-        when(bookSearchMock.search(givenQuery)).thenReturn(Collections.emptyList());
+        when(bookServiceMock.searchOneKeyword(givenQuery.toLowerCase())).thenReturn(getBookDtos());
         List<BookDto> givenSearchResults = bookRestAPIController.searchResults(givenQuery);
 
-        assertEquals(Collections.emptyList(), givenSearchResults);
+        assertEquals(2, givenSearchResults.size());
     }
```
Comment (patkkka) [inline springapp/src/test/java/com/javaacademy/robot/controllers/BookRestAPIControllerTest.java]: In all subsequent tests you have formatting convention to leave one blank line between 'given', 'when' and 'then' section. In all previous tests you have it only before assertion. You need to be consistent :)

**Pull request merged**